tests/typecheck-tests.ts

```diff
@@ -542,7 +542,72 @@ import { ok, err, okAsync, errAsync, Result, ResultAsync } from '../src'
 
 
 (function describe(_ = 'Combine on Unbounded lists') {
-  // TODO:
-  // https://github.com/supermacro/neverthrow/issues/226
+  (function describe(_ = 'combine') {
+    (function it(_ = 'combines different results into one') {
+      type Expectation = Result<[ number, string ], Error | string[]>;
+
+      const result: Expectation = Result.combine([
+        ok(1),
+        ok('string'),
+        err([ 'string', 'string2' ]),
+        err(new Error('error content')),
+      ]);
+    });
+
+    (function it(_ = 'combines only ok results into one') {
+      type Expectation = Result<[ number, string ], never>;
+
+      const result: Expectation = Result.combine([
+        ok(1),
+        ok('string'),
+      ]);
+    });
+
+    (function it(_ = 'combines only err results into one') {
+      type Expectation = Result<never, number | string>;
+
+      const result: Expectation = Result.combine([
+        err(1),
+        err('string'),
+      ]);
+    });
+
+    (function it(_ = 'combines empty list results into one') {
+      type Expectation = Result<never, never>;
+
+      const result: Expectation = Result.combine([]);
+    });
+  });
+
+  (function describe(_ = 'combineWithAllErrors') {
+    (function it(_ = 'combines different results into one') {
+      type Expectation = Result<[ number, string ], (Error | string[])[]>;
+
+      const result: Expectation = Result.combineWithAllErrors([
+        ok(1),
+        ok('string'),
+        err([ 'string', 'string2' ]),
+        err(new Error('error content')),
```

Comment:
Shouldn't the expectation here be:

 ```typescript
Result<[ number, string, never, never ], (Error | string[])[]>```

??

Reply:
I have not considered `never` types to be in order, therefore not processed them. When I'm free, I'll go back and add those and it will work. The hardest part is already done as you see. The simplest thing would be taking this result and the parameter array (which is readonly) and until their lengths matches, adding `never`s to array is the first solution I have in my mind but as I said, I'll see this when I'm free. Should not be a big problem as we could done it until this point.

Reply:
Updated the typing, see the commit.